internal/exec/engine.go

```diff
@@ -86,25 +93,12 @@ func (e Engine) Run(stageName string) error {
 	if err != nil && err != providers.ErrNoProvider {
 		e.Logger.Crit("failed to acquire system base config: %v", err)
 		return err
-	}
-
-	baseConfigFound := true
-	if err == providers.ErrNoProvider {
-		baseConfigFound = false
-	}
-
-	configExists, err := executil.PathExists(e.ConfigCache)
-	if err != nil {
-		e.Logger.Info("failed to check for %s : %v", e.ConfigCache, err)
-	}
-	// This is a hack to emit the journald message
-	// only once for the base config. Change this when
-	// (https://github.com/coreos/ignition/issues/950)
-	// is implemented.
-	if !configExists && baseConfigFound {
-		if err := logStructuredJournalEntry("base", "system", false); err != nil {
-			e.Logger.Info("failed to log systemd journal entry: %v", err)
-		}
+	} else if err == nil {
```

Comment:
Nit: I'd rather handle the `err == providers.ErrNoProvider` case somehow if needed (logging / bubbling-up?) and then un-indent this block out of the `else` branch.

Reply:
Hmm not sure I follow. From the POV of Ignition, `ErrNoProvider` is totally fine; it just means the OS doesn't ship a base config. I don't think we'd be able to un-indent this block somehow without making it conditional on "optional base config was found".

Reply:
I see, wrong suggestion then (TBH I'm internally starting to question the usage of error value for "totally fine" codepaths).